server/src/controllers/GrailController.ts

```diff
@@ -86,8 +102,157 @@ export class GrailController {
       dataToSet.runewordData = runewordGrailData;
       return dataToSet;
     });
+
+    this.getMemberParties(address, parties => {
+      parties.forEach(party => {
+        this.saveGrailInPartyDatabase(address, party.address, grailData);
+      });
+    });
   };
 
+  public saveGrailInPartyDatabase = async (
+    grailAddress: string,
+    partyAddress: string,
+    data: any
+  ) => {
+    let partyUserData = GrailController.formatGrailForParty(data);
+    const result = await this.partyCollection.findOneAndUpdate(
+      {
+        address: GrailController.trimAndToLower(partyAddress),
+        "data.users.username": grailAddress
+      },
+      {
+        $set: {
+          "data.users.$.data": partyUserData
+        },
+        $inc: { updateCount: 1 }
+      },
+      { returnOriginal: false }
+    );
+  };
+
+  public static formatGrailForParty = (data: any): any => {
+    let partyGrailData = {
+      uniqueArmor: {
+        missing: 123
+      },
+      uniqueWeapons: {
+        missing: 197
+      },
+      uniqueOther: {
+        missing: 59
+      },
+      sets: {
+        missing: 127
+      },
+      itemScore: 0
+    };
+    if (data && data.uniques) {
```

Comment:
As I already said once: I think it would be better if we do the whole score calculation on the client. By doing this we remove the bottleneck on the server because it's don on the client. Also we already do a big part of the calculation anyway.
However this can also be done in the future => issue

Reply:
https://github.com/Nasicus/d2-holy-grail/issues/30